website/docs/reference/warehouse-profiles/bigquery-profile.md

```diff
@@ -195,7 +195,97 @@ my-profile:
       priority: interactive
 ```
 
-### Timeouts
+### Timeouts and Retries
+
+<VersionBlock firstVersion="1.1">
+
+Every BigQuery query is made in two steps:
+1. Submit the query job to BQ server and obtain a query job id. 
+2. Wait for the query to execute and poll the query result. 
+
+dbt supports configuring BigQuery query timeouts in each step, as well as the retry and deadline of the overall query. 
+
+#### job_execution_timeout_seconds
+
+The timeout setting of step 2, `job_execution_timeout_seconds`, is the most common one.
+
+:::info Renamed config
+
+In older versions, this same config was just called `timeout_seconds`.
+
+:::
+  
+By default, the query execuion timeout is set to 300 seconds. If a dbt model takes longer than this timeout to complete, then BigQuery may cancel the query and issue the following error:
+
+```
+ Operation did not complete within the designated timeout.
+```
+  
+To change this timeout, use the `job_execution_timeout_seconds` configuration:
+
+```yaml
+my-profile:
+  target: dev
+  outputs:
+    dev:
+      type: bigquery
+      method: oauth
+      project: abc-123
+      dataset: my_dataset
+      job_execution_timeout_seconds: 600 # 10 minutes
+```
+
+#### job_creation_timeout_seconds
+  
+In addition, dbt also allow more advanced timeout control in step 1 via `job_creation_timeout_seconds`, in which dbt submits a query job to BQ JobInsert API server and receives a query job id in return. This step shall be very quick, usually under a few seconds. However, in some rare situation, it could take much longer to complete. 
```

Comment:
```suggestion
You can change the timeout seconds in the creation step by configuring  `job_creation_timeout_seconds`. The creation step is when dbt submits a query job to the BQ JobInsert API server and receives a query job ID in return. This step usually takes under a few seconds. However, in some rare situations, it could take longer to complete. 
```